core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
+    .orElse(Try(spark.sessionState.sqlParser.parsePlan("vacuum delta_table")).toOption)
+    .isEmpty ||
+    spark.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
+    throw new DeltaRuntimeException(
```

Comment:
Could you define this in `DeltaErrors`? For this case, it's better to use `DeltaIllegalStateException`.

Reply:
We already have `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` defined in DeltaErrors. Should we do this then?

Reply:
We can change `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` to `def configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` and call it.

Reply:
Sounds good. Here's the implementation:
```scala
 def configureSparkSessionWithExtensionAndCatalog(
      originalException: Option[Throwable]): Throwable = {
    val catalogImplConfig = SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key
    new DeltaIllegalStateException(
      errorClass = "DELTA_CONFIGURE_SPARK_SESSION_WITH_EXTENSION_AND_CATALOG",
      messageParameters = Array(classOf[DeltaSparkSessionExtension].getName,
        catalogImplConfig, classOf[DeltaCatalog].getName),
      cause = originalException.getOrElse(null))
  }
```